cli: fix 'shell' when under the irb gem

doudou opened the pull request:
Shell-related test started failing in CI because Readline was not defined. It turned out that the shell was not compatible with the gemified irb, which was pulled in some workspaces by iruby.

This pull request fixes the problems related to this interdependency.

gabrielfpacheco commented:
Hey, thanks for addressing that so quickly. I noticed this issue in the end of last week while running `syskit shell`, but didn't have the time to open an issue.

However, the suggested change in this PR does solve the `readline` dependency problem but leads to this one:

```console
Traceback (most recent call last):
	6: from /home/squidbot/dev/squidbot/tools/syskit/bin/syskit:69:in `<main>'
	5: from /home/squidbot/dev/squidbot/tools/roby/lib/roby/custom_require.rb:25:in `require'
	4: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	3: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	2: from /home/squidbot/dev/squidbot/tools/roby/lib/roby/app/scripts/shell.rb:161:in `<top (required)>'
	1: from /home/squidbot/dev/squidbot/tools/roby/lib/roby/app/scripts/shell.rb:161:in `new'
/home/squidbot/.local/share/autoproj/gems/ruby/2.5.0/gems/irb-1.3.5/lib/irb/workspace.rb:102:in `initialize': undefined method `source_location' for #<Binding:0x00005619ced83b08> (NoMethodError)
```

Any ideas?

doudou commented:
> /home/squidbot/.local/share/autoproj/gems/ruby/2.5.0/gems/irb-1.3.5/lib/irb/workspace.rb

Ah.

That's probably the actual issue. We are somehow pulling the irb gem instead of using the built-in.

doudou commented:
Do you guys have the iruby gem installed (from rock.jupyter) ?

doudou commented:
Seems that the remaining issue is 2.5 specific. Not happening in 2.7, but is happening on our 2.5-based CI as well.

doudou commented:
So, clearly related to the irb gem, and that's pulled by the newest version of iruby (released end of may).

I've added some commits (and rewrote the PR description) to match these findings.

doudou commented:
@gabrielfpacheco @g-arjones could you guys have a quick look at the last version ASAP ? Thanks.

gabrielfpacheco commented:
> could you guys have a quick look at the last version ASAP ? Thanks.

Now it works just fine to me! Thank you

g-arjones commented on lib/roby/app/scripts/shell.rb:
```diff
@@ -103,7 +104,7 @@ def gets
 
-class ShellEvalContext < BasicObject
+class ShellEvalContext
     include ::Kernel
```
Is there a reason to keep including `Kernel` now that `ShellEvalContext` is not just a basic object anymore?

doudou replied:
Nope. Good catch.
